src/test/test-parse-helpers.c

```
@@ -92,4 +94,41 @@ TEST(invalid_items) {
         test_invalid_item("ipv6:tcp:6666\n zupa");
 }
 
+TEST(open_file_parse) {
+        _cleanup_(open_file_freep) OpenFile *of = NULL;
+        int r;
+
+        r = open_file_parse("/proc/1/ns/mnt:host-mount-namespace:ro", &of);
+
+        assert_se(r == 0);
+        assert_se(streq(of->path, "/proc/1/ns/mnt"));
+        assert_se(streq(of->fdname, "host-mount-namespace"));
+        assert_se(of->flags == OPENFILE_RDONLY);
+
+        r = open_file_parse("/proc/1/ns/mnt", &of);
+
+        assert_se(r == 0);
+        assert_se(streq(of->path, "/proc/1/ns/mnt"));
+        assert_se(streq(of->fdname, "/proc/1/ns/mnt"));
+        assert_se(of->flags == 0);
+
+        r = open_file_parse("/proc/1/ns/mnt:host-mount-namespace", &of);
+
+        assert_se(r == 0);
+        assert_se(streq(of->path, "/proc/1/ns/mnt"));
+        assert_se(streq(of->fdname, "host-mount-namespace"));
+        assert_se(of->flags == 0);
+
+        r = open_file_parse("/proc/1/ns/mnt::rw", &of);
```

We should fail if unknown options such as `rw` are passed